Community/gitlab_import/gitlab_import_page.py

```diff
@@ -0,0 +1,213 @@
+                    print("Found utils to load")
+                    # Is the folder really there?
+                    if os.path.exists(gitlab_import_config.gw_utils_directory):
+                        # Move folder from zip to deployed util dir
+                        # backup folder
+                        # Does the backup folder exist
+                        if os.path.exists(gitlab_import_config.backups_folder):
+
+                            # Create the backup file
+                            shutil.make_archive(
+                                gitlab_import_config.backups_folder + '/' + gitlab_import_config.gitlab_import_utils_directory + '-' + time_format,
+                                'zip', gitlab_import_config.gw_utils_directory)
+                        else:
+                            # Create backup folder
+                            os.mkdir(gitlab_import_config.backups_folder)
+                            shutil.make_archive(
+                                gitlab_import_config.backups_folder + '/' + gitlab_import_config.gitlab_import_utils_directory + '-' + time_format,
+                                'zip', gitlab_import_config.gw_utils_directory)
+                        # Delete the util folder on the GW server
+                        shutil.rmtree(gitlab_import_config.gw_utils_directory)
+                        # Move GitHub Util folder to GW server
+                        shutil.move(
+                            os.path.join(gitlab_import_config.workflow_dir, folder_name, gitlab_import_config.gitlab_import_utils_directory),
+                            gitlab_import_config.gw_utils_directory.rsplit('/', 1)[0])
+                    else:
+                        # Move folder from zip to deployed util dir
+                        shutil.move(
+                            os.path.join(gitlab_import_config.workflow_dir, folder_name, gitlab_import_config.gitlab_import_utils_directory),
+                            gitlab_import_config.gw_utils_directory.rsplit('/', 1)[0])
+
+                else:
+                    print("No utils specified in custom.gitlab_import_utils_directory")
+
+            except Exception as e:
+                app.logger.exception("Failed to load GitLab Util folder: {}".format(str(e)))
+                g.user.logger.warning('%s' % util.safe_str(e), msg_type=g.user.logger.EXCEPTION)
+                flash(
+                    'Failed to load GitLab Util folder properly. Check to see if path ' + gitlab_import_config.gw_utils_directory + ' exsists on the server.')
+                return redirect(url_for('gitlab_importgitlab_import_gitlab_import_page'))
+
+            # Create backup
+            shutil.make_archive(
+                gitlab_import_config.backups_folder + '/' + gitlab_import_config.gitlab_import_directory + '-' + time_format,
+                'zip', os.path.join(gitlab_import_config.workflow_dir, gitlab_import_config.gitlab_import_directory))
```

Comment:
you might also want to make a backup of the ps folder in case customizations screws up their environment